vault/token_store.go

```diff
@@ -2154,7 +2154,14 @@ func (ts *TokenStore) handleCreateCommon(ctx context.Context, req *logical.Reque
 	tokenTypeStr := data.Type
 	if role != nil {
 		switch role.TokenType {
-		case logical.TokenTypeDefault, logical.TokenTypeService:
+		case logical.TokenTypeDefault, logical.TokenTypeDefaultService:
+			// Use the user-given value, but fall back to service
+		case logical.TokenTypeDefaultBatch:
+			// Use the user-given value, but fall back to batch
+			if tokenTypeStr == "" {
+				tokenTypeStr = logical.TokenTypeBatch.String()
+			}
+		case logical.TokenTypeService:
```

Comment:
Think we should error if the requested tokenTypeStr != the configured type?

Reply:
I think we should ignore it. Think of e.g. the Nomad case where a client may *request* a certain type of token but get overridden by the admin -- probably it'd be better for the job to simply work (but with a service token) than fail. WDYT?